instrumentation/httpurlconnection/README.md

```diff
@@ -0,0 +1,75 @@
+- Traces and spans are generated and properly closed.
+- Any exceptions thrown are recorded.
+
+A span associated with a given request is concluded in the following scenarios:
+- When the getInputStream()/getErrorStream() APIs are called, the span concludes after the stream is fully read, an IOException is encountered, or the stream is closed.
+- When the disconnect API is called.
+
+Spans won't be automatically ended and reported otherwise. If any of your URLConnection requests do not call the span concluding APIs mentioned above, refer the section entitled ["Scheduling Harvester Thread"](#scheduling-harvester-thread). This section provides guidance on setting up a recurring thread that identifies unreported, idle connections (those that have been read from but have been inactive for more than 10 seconds) and concludes any open spans associated with them.
+
+> The minimum supported Android SDK version is 21, though it will also instrument APIs added in the Android SDK version 24 when running on devices with API level 24 and above.
+
+> If your project's minSdk is lower than 26, then you must enable
+> [corelib desugaring](https://developer.android.com/studio/write/java8-support#library-desugaring).
+>
+> If your project's minSdk is lower than 24, in order to run the app built on debug, you need to add the following property in `gradle.properties` file:
+> -  If AGP <= 8.3.0, set `android.enableDexingArtifactTransform=false`
+> - if AGP > 8.3.0, set `android.useFullClasspathForDexingTransform=true`
+>
+> For the full context for these workaround, please see
+> [this issue](https://issuetracker.google.com/issues/334281968) for AGP <= 8.3.0
+> or [this issue](https://issuetracker.google.com/issues/230454566#comment18) for AGP > 8.3.0.
+
+### Add these dependencies to your project
+
+Replace `AUTO_HTTP_URL_INSTRUMENTATION_VERSION` with the [latest release](https://central.sonatype.com/search?q=g%3Aio.opentelemetry.android++a%3Ahttpurlconnection-library&smo=true).
+
+Replace `BYTEBUDDY_VERSION` with the [latest release](https://search.maven.org/search?q=g:net.bytebuddy%20AND%20a:byte-buddy).
+
+#### Byte buddy compilation plugin
+
+This plugin leverages Android's [Transform API](https://developer.android.com/reference/tools/gradle-api/current/com/android/build/api/variant/ScopedArtifactsOperation#toTransform(com.android.build.api.artifact.ScopedArtifact,kotlin.Function1,kotlin.Function1,kotlin.Function1)) to instrument bytecode at compile time. You can find more info on its [repo page](https://github.com/raphw/byte-buddy/tree/master/byte-buddy-gradle-plugin/android-plugin).
+
+```groovy
+plugins {
+    id 'net.bytebuddy.byte-buddy-gradle-plugin' version 'BYTEBUDDY_VERSION'
+}
+```
+
+#### Project dependencies
+
+```kotlin
+implementation("io.opentelemetry.android:httpurlconnection-library:AUTO_HTTP_URL_INSTRUMENTATION_VERSION")
+byteBuddy("io.opentelemetry.android:httpurlconnection-agent:AUTO_HTTP_URL_INSTRUMENTATION_VERSION")
```

Comment:
We have release automation now that can update these versions with the current/latest (see `prepare-release-branch.yml`). It doesn't have to be part of this PR, but we should probably open an issue to automate this as part of the release.

Doing that would also allow us to remove the statement above that tells the reader to replace a thing.

Reply:
Thanks for sharing that! That would be better. Created this issue to handle that - https://github.com/open-telemetry/opentelemetry-android/issues/417.